code/modules/power/smes.dm

```diff
@@ -262,9 +262,10 @@
 		if(outputting)
 			output_used = min( charge/SMESRATE, output_level)		//limit output to that stored
 
-			charge -= output_used*SMESRATE		// reduce the storage (may be recovered in /restore() if excessive)
-
-			add_avail(output_used)				// add output to powernet (smes side)
+			if (add_avail(output_used))				// add output to powernet if it exists (smes side)
+				charge -= output_used*SMESRATE		// reduce the storage (may be recovered in /restore() if excessive)
+			else
+				outputting = 0
```

Comment:
outputting = FALSE.
While you're at it, change all values that represent true and false as 1's and 0's to TRUE and FALSE

Reply:
Sorted